### PR TITLE
add Repitools extension to R-bundle-Bioconductor 3.9 (and take into extension version in exts_filter)

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
generoso - Linux centos linux 7.6.1810, Intel(R) Xeon(R) CPU E5-2660 v4 @ 2.00GHz, Python 3.6.8
See https://gist.github.com/97f17d11fe2a0b2de589e1b8c049a0cd for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2404.golett.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz, Python 2.7.5
See https://gist.github.com/b1de3adb3aec15bc1753ec87ab5bb0dc for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node3169.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/f08870ff7c7e7539f91e23bb5ad67cba for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node3169.skitty.os - Linux centos linux 7.7.1908, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/8204cbac3229e7e352648e58ccc967b7 for a full test report.

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node368.hydra.os - Linux centos linux 7.6.1810, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz, Python 2.7.5
See https://gist.github.com/7a03df545e4c203b022ce12912e27fb5 for a full test report.

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.9-foss-2019a-R-3.6.0.eb [hunk 1]
```diff
@@ -31,7 +31,12 @@ exts_default_options = {
 }
 
 exts_defaultclass = 'RPackage'
-exts_filter = ("R -q --no-save", "library(%(ext_name)s)")
+
+# check whether correct version is installed in extension filter
+# (some versions in this bundle may be newer than the ones provided by R)
+local_ext_version_check = "if (packageVersion('%(ext_name)s') != '%(ext_version)s')"
+local_stop_msg = "stop('%(ext_name)s %(ext_version)s not installed')"
```
Comment (smoors) [inline easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.9-foss-2019a-R-3.6.0.eb]: @boegel maybe can we differentiate the message here?
between packages that have not been installed and packages that have been installed but with the wrong version

Comment (boegel) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.9-foss-2019a-R-3.6.0.eb]: @smoors That's already the case. If a package is not installed, `packageVersion` will fail, and the `stop` will not be run.

Basically, this:

```
# non-existing package
$ echo "if (packageVersion('foo') != '1.2.3') stop('foo 1.2.3 not installed')" | R -q --no-save
> if (packageVersion('foo') != '1.2.3') stop('foo 1.2.3 not installed')
Error in packageVersion("foo") : there is no package called ‘foo’
Execution halted
$ echo $?
1
```

```
# installed package but different version
$ echo "if (packageVersion('R.utils') != '1.2.3') stop('R.utils 1.2.3 not installed')" | R -q --no-save
> if (packageVersion('R.utils') != '1.2.3') stop('R.utils 1.2.3 not installed')
Error: R.utils 1.2.3 not installed
Execution halted
$ echo $?
1
```

**Comment by verdurin** [comment]
Test report by @verdurin
**FAILED**
Build succeeded for 1 out of 2 (1 easyconfigs in this PR)
easybuild.novalocal - Linux centos linux 7.7.1908, Intel Xeon Processor (Skylake, IBRS), Python 2.7.5
See https://gist.github.com/d9065ac0a75c1eac29d337fc9992cc23 for a full test report.

**Comment by verdurin** [comment]
Here's my build failure:
```
== 2019-11-29 16:15:28,716 rpackage.py:225 DEBUG Installing R package MLInterfaces version 1.64.0.
== 2019-11-29 16:15:28,716 rpackage.py:152 DEBUG make_cmdline_cmd returns  R CMD INSTALL /eb/maint/sources/r/R-bundle-Bioconductor/extensions/MLInterfaces_1.64.0.tar.gz   --library=/eb/maint/software/R-bundle-Bioconductor/3.9-foss-2019a-R-3.6.0/ --no-clean-on-error
== 2019-11-29 16:15:28,716 run.py:200 DEBUG run_cmd: running cmd  R CMD INSTALL /eb/maint/sources/r/R-bundle-Bioconductor/extensions/MLInterfaces_1.64.0.tar.gz   --library=/eb/maint/software/R-bundle-Bioconductor/3.9-foss-2019a-R-3.6.0/ --no-clean-on-error  (in /tmp/eb-2acLcP/tmpBGdH44/easybuilders)
== 2019-11-29 16:15:28,716 run.py:219 INFO running cmd:  R CMD INSTALL /eb/maint/sources/r/R-bundle-Bioconductor/extensions/MLInterfaces_1.64.0.tar.gz   --library=/eb/maint/software/R-bundle-Bioconductor/3.9-foss-2019a-R-3.6.0/ --no-clean-on-error
== 2019-11-29 16:15:29,085 build_log.py:164 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): cmd " R CMD INSTALL /eb/maint/sources/r/R-bundle-Bioconductor/extensions/MLInterfaces_1.64.0.tar.gz   --library=/eb/maint/software/R-bundle-Bioconductor/3.9-foss-2019a-R-3.6.0/ --no-clean-on-error " exited with exit code 1 and output:
ERROR: dependencies 'rda', 'threejs' are not available for package 'MLInterfaces'
 (at easybuild/tools/run.py:529 in parse_cmd_output)
```

**Comment by boegel** [comment]
@verdurin You need to add the missing extensions in your `R` installation using `eb --force --skip`, see https://github.com/easybuilders/easybuild-easyconfigs/pull/8881

File: easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.9-foss-2019a-R-3.6.0.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/r/R-bundle-Bioconductor/R-bundle-Bioconductor-3.9-foss-2019a-R-3.6.0.eb]: Hmm, perhaps this is better though:
```
$ echo "pkgver = packageVersion('R.utils'); if (pkgver != '1.2.3') stop('R.utils 1.2.3 not installed, found ', pkgver)" | R -q --no-save
> pkgver = packageVersion('R.utils'); if (pkgver != '1.2.3') stop('R.utils 1.2.3 not installed, found ', pkgver)
Error: R.utils 1.2.3 not installed, found 2.8.0
Execution halted
```

fixed in 8b91562

**Comment by verdurin** [comment]
Test report by @verdurin
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
easybuild.novalocal - Linux centos linux 7.7.1908, Intel Xeon Processor (Skylake, IBRS), Python 2.7.5
See https://gist.github.com/06a4527cfe8e6bfa0c6ba104c5447795 for a full test report.

**Comment by verdurin** [comment]
Going in, thanks @boegel!

**Comment by akesandgren** [comment]
Building this gives me:
ERROR: dependency idr is not available for package ChIPpeakAnno
And, as far as I can see, idr is not part of the updated R/3.6.0 from #8881 either

**Comment by boegel** [comment]
@akesandgren `idr` is included in the `R` 3.6.0 easyconfigs in `develop`, was added via #9184

**Comment by akesandgren** [comment]
Ok, this one didn't mention that one...

**Pull request merged**